src/contract/mod.rs

```diff
@@ -284,51 +284,75 @@ impl DocumentType {
                 "unable to get document properties from the contract",
             )))?;
 
-        // Based on the property name, determine the type
-        for (property_key, property_value) in property_values {
+        fn insert_values(document_properties: &mut HashMap<String, types::DocumentFieldType>, prefix: Option<String>, property_key: &Value, property_value: &Value) -> Result<(), Error> {
             if !property_key.is_text() {
                 return Err(Error::CorruptedData(String::from(
                     "property key should be text",
                 )));
             }
 
+            let property_key_string = property_key.as_text().expect("confirmed as text").to_string();
+
+            let prefixed_property_key = match prefix {
+                None => {property_key_string}
+                Some(prefix) => { vec![prefix, property_key_string].join(".")}
```

Comment:
```suggestion
                None => property_key_string,
                Some(prefix) => format!("{prefix}.{property_key_string}"),
```

Reply:
It's better to use the first option becuase we get to move, instead of having to copy when using format!

Reply:
Using fist option we allocate another vector, then allocate another string using join

Reply:
Okay, I completely agree, we should not be using vec! here and allocating a vector. I think the better solution would be smth like
```suggestion
                None => {property_key_string}
                Some(prefix) => { [prefix, property_key_string].join(".")}
```